static/js/copy_and_paste.js

```diff
@@ -232,18 +232,16 @@ export function analyze_selection(selection) {
         }
 
         $endc = $(range.endContainer);
-        // If the selection ends in the bottom whitespace, we should
-        // act as though the selection ends on the final message.
-        // This handles the issue that Chrome seems to like selecting
-        // the compose_close button when you go off the end of the
-        // last message
-        if ($endc.attr("id") === "bottom_whitespace" || $endc.attr("id") === "compose_close") {
-            $initial_end_tr = $(".message_row").last();
-            // The selection goes off the end of the message feed, so
-            // this is a multi-message selection.
-            skip_same_td_check = true;
+        // Sometimes (especially when three click selecting in Chrome) the selection
+        // can end in the bottom whitespace, the compose close button, or an unselectable
+        // area of the next message like the unread counter. Ideally we make it so those
+        // things aren't selectable, but for now we just find the message row directly above
+        // wherever the selection ended. If the node that is selected is a text node, we're
+        // already in the right place.
+        if ($endc[0].nodeName === "#text") {
+            $initial_end_tr = $($endc.parents(".message_row")[0]);
         } else {
-            $initial_end_tr = $($endc.parents(".selectable_row")[0]);
```

Comment:
Why the `selectable_row` => `message_row` change? IIRC, the reason the `selectable_row` CSS class exists is so that we can have other types of rows in the message feed that are treated as a message row for the purpose of copy/paste (I think date dividers might be an example).

It looks like we don't actually have any such rows at present, but it still seems weird to change this detail?

Reply:
Yeah, this is part of my above question, which could have been phrased better:

Why do we make `initial_end_tr` a `selectable_row` here but a `message_row` in the first if statement? I switched them all to `message_row` to be consistent, but I don't remember why I thought that was the correct choice :p

Reply:
I've switched it back, but I'm still curious about the answer to that question.

Reply:
I'm not sure. If I had to guess, these are intended to be all` .selectable_row`, but of course we don't have any other selectable rows right now, so it doesn't matter.